src/main/java/io/confluent/connect/elasticsearch/DataConverter.java

```diff
@@ -130,10 +131,13 @@ static Schema preProcessSchema(Schema schema) {
         Schema valueSchema = schema.valueSchema();
         String keyName = keySchema.name() == null ? keySchema.type().name() : keySchema.name();
         String valueName = valueSchema.name() == null ? valueSchema.type().name() : valueSchema.name();
+        if (keySchema.type() == Schema.Type.STRING) {
+          return SchemaBuilder.map(preProcessSchema(keySchema), preProcessSchema(valueSchema)).build();
+        }
         Schema elementSchema = SchemaBuilder.struct().name(keyName + "-" + valueName)
-            .field(MAP_KEY, preProcessSchema(keySchema))
-            .field(MAP_VALUE, preProcessSchema(valueSchema))
-            .build();
+                .field(MAP_KEY, preProcessSchema(keySchema))
```

Comment:
nit: Sorry to return to this, but seems that the previous setting was the correct one re: indentation in this project. Tab needs to correspond to 2 spaces, thus this should be 4 spaces instead of 8 after the change. I wouldn't mind fixing bad indentation but the former was the correct one.